packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1433,6 +1439,15 @@ class StackViewport extends Viewport implements IStackViewport {
     this.imageIds = imageIds;
     this.currentImageIdIndex = currentImageIdIndex;
     this.targetImageIdIndex = currentImageIdIndex;
+    const imageRetrieveConfiguration = metaData.get(
+      imageRetrieveMetadataProvider.IMAGE_RETRIEVE_CONFIGURATION,
+      imageIds[currentImageIdIndex],
+      'stack'
+    );
+
+    this.retrieveConfiguration = imageRetrieveConfiguration?.constructor
```

Comment:
rename maybe `this.imageRetriever`

Reply:
imagesLoader since it does the full load, and you wanted the function to be imagesLoad